File: Sources/BiangBiangUI/Services/HistoryEntry.swift
import Foundation

public struct HistoryEntry: Identifiable, Codable, Equatable, Sendable {
    public let id: UUID
    public let original: String
    public let transliteration: String
    public let variantId: String
    public let timestamp: Date

    public init(
        id: UUID = UUID(),
        original: String,
        transliteration: String,
        variantId: String,
        timestamp: Date = Date()
    ) {
        self.id = id
        self.original = original
        self.transliteration = transliteration
        self.variantId = variantId
        self.timestamp = timestamp
    }
}

File: android/biangbiang-ui/src/main/java/dev/veeso/biangbiangui/services/HistoryEntry.kt
package dev.veeso.biangbiangui.services

import java.util.UUID

/**
 * A single transliteration entry persisted in history.
 *
 * Mirrors iOS `HistoryEntry`. The reference Android app keyed entries by a
 * `HistoryVariant` enum (MANDARIN/CANTONESE); the library generalises this to
 * an opaque [variantId] string so any [dev.veeso.biangbiangui.config.LanguageVariant]
 * can be recorded without an app-specific enum.
 */
data class HistoryEntry(
    val id: String = UUID.randomUUID().toString(),
    val original: String,
    val transliteration: String,
    val variantId: String,
    val timestamp: Long = System.currentTimeMillis(),
)
